setup.go

```diff
@@ -319,17 +322,39 @@ func Setup(ctx context.Context, cfg Config, key crypto.PrivKey, dnsCache *cached
 		}
 	}
 
+	var pbrf bsserver.PeerBlockRequestFilter
+	if cfg.PeeringCache && len(cfg.Peering) > 0 {
+		peers := make(map[peer.ID]struct{}, len(cfg.Peering))
+		for _, a := range cfg.Peering {
+			peers[a.ID] = struct{}{}
+		}
+
+		pbrf = func(p peer.ID, c cid.Cid) bool {
+			_, ok := peers[p]
+			return ok
+		}
+	} else {
+		pbrf = func(p peer.ID, c cid.Cid) bool {
+			return false
+		}
+	}
+
 	bsctx := metri.CtxScope(ctx, "ipfs_bitswap")
 	bn := bsnet.NewFromIpfsHost(h, router)
-	bswap := bsclient.New(bsctx, bn, blkst,
+	bswap := bitswap.New(bsctx, bn, blkst,
+		// --- Client Options
 		// default is 1 minute to search for a random live-want (1
 		// CID).  I think we want to search for random live-wants more
 		// often although probably it overlaps with general
 		// rebroadcasts.
-		bsclient.RebroadcastDelay(delay.Fixed(10*time.Second)),
+		bitswap.RebroadcastDelay(delay.Fixed(10*time.Second)),
 		// ProviderSearchDelay: default is 1 second.
-		bsclient.ProviderSearchDelay(time.Second),
-		bsclient.WithoutDuplicatedBlockStats(),
+		bitswap.ProviderSearchDelay(time.Second),
+		bitswap.WithoutDuplicatedBlockStats(),
+
+		// ---- Server Options
+		bitswap.WithPeerBlockRequestFilter(pbrf),
+		bitswap.ProvideEnabled(true),
 	)
```

Comment:
Another option here would be to create only a client with `bsclient.New` if peering cache is disabled. But to me it seemed that a peer block request filter that always returns `false` should do the same thing. Depends a bit more on the internals of the bitswap package.